src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CodeActionEndpoint.cs

```diff
@@ -176,8 +176,12 @@ internal async Task<RazorCodeActionContext> GenerateRazorCodeActionContextAsync(
         private async Task<IEnumerable<RazorCodeAction>> GetCSharpCodeActionsAsync(RazorCodeActionContext context, CancellationToken cancellationToken)
         {
             var csharpCodeActions = await GetCSharpCodeActionsFromLanguageServerAsync(context, cancellationToken);
-            var filteredCSharpCodeActions = await FilterCSharpCodeActionsAsync(context, csharpCodeActions, cancellationToken);
+            if (csharpCodeActions is null || !csharpCodeActions.Any())
```

Comment:
What was the behavior in this case before? Does it just take longer because we run through each provider with a null or empty list?

Reply:
It was causing an error later in the code action provider as it expected >0 code actions. It's also a short circuit as you suggested as we don't have to run through the providers which would all return empty results as we don't have any code actions to process.